org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/scoping/batch/AbstractSessionBasedScope.java

```diff
@@ -108,6 +109,43 @@ protected String toProperty(String methodName, JvmFeature feature, int getterPar
 		return PropertyUtil.getPropertyName(feature, methodName, getterParams, setterParams);
 	}
 	
+	/**
+	 * Returns the name as a property name, e.g. a prefix {@code get}, {@code is} or {@code set}
+	 * can be used with the result of this method.
+	 * If the given name is invalid, the result is <code>null</code>.
+	 */
+	/* @Nullable */
+	protected String tryGetAsPropertyName(String name) {
```

Comment:
@szarnekow I reverted this method as it did not do what I thought it did.
The method checks if a given name is valid to be extended (add set/get/is) to it. e.g. Class.CanonicalName is not valid as it doesn't map to Class.getCanonicalName. I added a new method in PropertyUtil that does actual conversion from full function names to their short hand name.